pipelines/hubspot/__init__.py

```diff
@@ -0,0 +1,68 @@
+from typing import Sequence, Iterator
+
+import dlt
+
+from dlt.common.typing import TDataItems
+from dlt.extract.source import DltResource
+
+from pipelines.hubspot.client import fetch_data
+from pipelines.hubspot.endpoints import CRM_CONTACTS_ENDPOINT, CRM_COMPANIES_ENDPOINT, \
+    CRM_DEALS_ENDPOINT, CRM_TICKETS_ENDPOINT, CRM_PRODUCTS_ENDPOINT, CRM_ENGAGEMENTS_CALLS_ENDPOINT, \
+    CRM_ENGAGEMENTS_COMMUNICATIONS_ENDPOINT, CRM_ENGAGEMENTS_EMAILS_ENDPOINT, WEB_ANALYTICS_EVENTS_ENDPOINT, CRM_QUOTES_ENDPOINT
+
+
+@dlt.source
+def hubspot(api_key: str = dlt.secrets.value) -> Sequence[DltResource]:
+    """
+    A DLT source that retrieves data from the HubSpot API using the specified API key.
+
+    This function retrieves data for several HubSpot API endpoints, including companies, contacts, deals,
+    tickets, products, engagements calls, engagements communications, engagements emails, and web analytics
+    events. It returns a tuple of Dlt resources, one for each endpoint.
+
+    Args:
+        api_key (str): The HubSpot API key to use for authentication. This can be passed as a string,
+            or can be fetched from a Dlt secrets store using `dlt.secrets.value`.
+
+    Returns:
+        tuple: A tuple of Dlt resources, one for each HubSpot API endpoint.
+
+    Notes:
+        This function uses the `fetch_data` function to retrieve data from the HubSpot CRM API. The API key
+        is passed to `fetch_data` as the `api_key` argument.
+
+        This function is decorated with `dlt.source`, which indicates that it is a Dlt source function.
+        When this function is included in a dlt pipeline, it will be executed to retrieve data for the
+        specified HubSpot API endpoints.
+
+    """
+
+    @dlt.resource(write_disposition="replace")
+    def companies() -> Iterator[TDataItems]:
```

Comment:
you could generate resources dynamically. see the `pipedrive` pipeline. OFC the list here is also good - it is a matter of taste (to some degree)

Reply:
Yes, that is true, but i think in this way is more visible that resources are included, i will suggest keep it like this, at least for now when the number of resources is till manageable